wandb/sdk/wandb_run.py

```diff
@@ -2931,20 +2932,19 @@ def _assert_can_log_artifact(self, artifact) -> None:  # type: ignore
 
     def _prepare_artifact(
         self,
-        artifact_or_path: Union[wandb_artifacts.Artifact, str],
+        artifact_or_path: Union[wandb_artifacts.Artifact, StrPath],
         name: Optional[str] = None,
         type: Optional[str] = None,
         aliases: Optional[List[str]] = None,
     ) -> Tuple[wandb_artifacts.Artifact, List[str]]:
-        if isinstance(artifact_or_path, str):
-            if name is None:
-                name = f"run-{self._run_id}-{os.path.basename(artifact_or_path)}"
-            artifact = wandb.Artifact(name, type)
+        if isinstance(artifact_or_path, (str, os.PathLike)):
+            name = name or f"run-{self._run_id}-{os.path.basename(artifact_or_path)}"
+            artifact = wandb.Artifact(name, type or "unspecified")
```

Comment:
I can't seem to find a mention of this in the backend, would you mind testing this against a real backend and making sure nothing breaks?

Reply:
I was wondering if this type string actually matches some sort of an enum in the backend, cuz if so, we could have broken the SDK <> backend integration. So I tried running a quick test, and I _think_ this is how the artifact will show up:
![Screen Shot 2023-03-15 at 11 52 43 AM](https://user-images.githubusercontent.com/1059168/225414030-4301cb59-6fdd-4c58-b513-e65f43ab7cb1.png)
SDK <> backend integration works well!

Reply:
No, I often just throw `type="as5eest"` or random garbage in while testing; there _are_ places we treat type as a special value, such as for `type="model"`, but `unspecified` isn't one of them.